pkg/kubernetes/client/v1alpha1/client.go

```diff
@@ -134,6 +135,21 @@ func New(opts ...OptionFn) *Client {
 	return c
 }
 
+var (
+	instance *Client
+	once     sync.Once
+)
+
+// Instance returns a singleton instance of
+// this client
+func Instance() *Client {
+	once.Do(func() {
+		instance = New()
```

Comment:
this being single instance to exists, should we check for NULL for instance before doing allocation?

Reply:
We are leaving this check to library. I believe library does it.
https://golang.org/src/sync/once.go

Reply:
sounds good :+1: